libs/wingc/src/type_check.rs

```diff
@@ -1124,7 +1124,19 @@ impl<'a> TypeChecker<'a> {
 		statement_idx: usize,
 		context: &TypeCheckerContext,
 	) -> TypeRef {
-		let t = match &exp.kind {
+		let t = self.type_check_exp_helper(&exp, env, statement_idx, context);
+		exp.evaluated_type.replace(Some(t));
+		t
+	}
+
+	fn type_check_exp_helper(
```

Comment:
I think the `type_check_exp_helper` name emphasizes that nobody else should be calling it except for `type_check_exp`. I'll add a comment to clarify. (It could have also been an inner function / closure, but I wanted to avoid the extra nesting)